services/QuillLMS/app/services/classlink_integration/user_persister.rb

```diff
@@ -30,6 +30,8 @@ def run
     end
 
     def link_existing_user_to_classlink
+      return migrate_to_new_instance(existing_user) if existing_user.classlink_user?
```

Comment:
Take or leave, since `existing_user` is a memoized method, you might consider not passing it as a param to just pass it back, e.g.
```ruby
if existing_user.classlink_user?
  migrate_to_new_instance
  return existing_user
end